examples/app_blink.rs

```diff
@@ -0,0 +1,37 @@
+#![feature(no_std, core, start)]
+#![no_std]
+
+extern crate core;
+extern crate zinc;
+
+use zinc::hal::timer::Timer;
+use zinc::hal::lpc17xx::{pin, timer};
+use zinc::hal::pin::GpioDirection;
+use zinc::hal::pin::Gpio;
+use core::option::Option::Some;
+
+#[start]
```

Comment:
I actually have a replacement for that in [mcoffin/remove-start](/mcoffin/zinc/tree/remove-start). I left it out of this PR for now to speed up adoption, but that simple syntax extension would generate a dummy version of `#[start]` for users not using platformtree.